### ISSUE TITLE
is it possible to get the length of video?

### ISSUE THREAD

**Issue by ApolloRoboto** [pytube/pytube#296]
Is it possible to get the length of a video before downloading? I am trying to limit all downloaded videos at 10 minutes. I have read the documents and tried this:
`from pytube import YouTube`
`yt = YouTube(url)`
`print(yt.length)`
but it keeps on giving me this error: AttributeError: 'YouTube' object has no attribute 'length'

**Comment by grking8** [comment]
would it be possible to get an example url?

I tried

```
from pytube import YouTube

url = 'https://www.youtube.com/watch?v=kHLHSlExFis'
yt = YouTube(url)
print('video description: ', yt.description)
print('rating', yt.rating)
print('length', yt.length)
print('views', yt.views)
```

output

```
video description:  God is a woman (Official Video)Song available here: https://arianagrande.lnk.to/GodIsAWom...Connect with Ariana: https://www.arianagrande.com https://www.instagram.com/arianagrande https://twitter.com/arianagrande https://www.facebook.com/arianagrande Video Director: Dave MeyersVideo Producer: Nathan ScherrerVideo Editor: Nick GilbergA FREENJOY PRODUCTION Body Painting by Alexa MeadeMadonna appears courtesy of Live Nation Worldwide, Inc. and Maverick Management/Guy Oseary.  Vocal Production and Engineering for Madonna by Lauren D’Elia.Music video by Ariana Grande performing God is a woman. © 2018 Republic Records, a Division of UMG Recordings, Inc.http://vevo.ly/dlM76Z
rating 4.57178401947
length 242
views 34985094
```

**Comment by ApolloRoboto** [comment]
Well, it just doesn't work with any url I've tried, including the one in your reply.
yt.description, yt.rating yt.length and yt.views doesn't work. however, yt.title does work

i tried to update the package with pip3 install -U pytube: nothing changed so i uninstalled/reinstall and tried again but still no result*
I also tried on a different machine

**Comment by grking8** [comment]
Given that `.title` works and the others don't, and the others were merged after the latest tag `9.2.2` (`pip install` will get you the code as of `9.2.2` release), looks like you might not be running off the latest code on `master`

Depending on your use case, you can update your `master` branch or wait for the next release to PyPI

**Comment by ApolloRoboto** [comment]
well, i guess i will just wait for a the next release. while waiting i will limit the download on the file size instead,

**Comment by aflawt** [comment]
try pip3 install https://github.com/nficano/pytube/archive/master.zip

**Comment by raveenb** [comment]
this problem seems to continue, 
the code for the length should read from, I see there is a Pull request already. Requesting maintainer to do a merge

```
 @property
    def length(self):
        """Get the video length in seconds.

        :rtype: str

        """

        return self.player_config_args["player_response"]["videoDetails"]['lengthSeconds']
```

**Comment by Jack-of-tg** [comment]
it's it possible to print the size of particular stream by pytube

**Comment by DavidARivkin** [comment]
I actually get a few errors:

-   yt.title throws an exception and says to report a bug for the "title" key.
-  yt.length reports TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'

Is there a planned update as it all seems very old.  PyTube3 does not work at all.

**Comment by Reno-Codes** [comment]
@DavidARivkin I found a workaround for the issues.

It happens because Pytube currently uses an outdated Youtube version, it actually interacts with youtube via defined clients, it uses an Android app client by default, but I found it more reliable to use a WEB client.

Find innertube.py, if anyone is struggling to find it, it's simple:

write "from pytube.innertube import InnerTube"
1. In my case, in VS code, hold CTRL while hovering above the word "InnerTube", click it and it will take you to innertube.py
2. Scroll up, you'll see "_default_clients", in first "WEB" client, change clientVersion to '2.20230427.04.00'
3. Scroll back down to "class Innertube", in _init_, you have "client='ANDROID'", just change it to client='WEB' 

And voila... :)
